python/test/unit/language/test_tlx.py

```diff
@@ -1259,6 +1259,93 @@ def remote_shmem_store_kernel(
     assert y[1] == 142.0 and y[0] == 243.0
 
 
+@pytest.mark.skipif(not is_blackwell(), reason="Need Blackwell")
+@pytest.mark.parametrize("num_ctas", [1, 2])
+def test_async_remote_shmem_store(num_ctas, device):
+    """Test that remote_shmem_store correctly aggregates 2D data across multiple CTAs."""
+
+    @triton.jit
+    def remote_store_sum_kernel(
+        input_ptr,
+        output_ptr,
+        M: tl.constexpr,
+        N: tl.constexpr,
+        BLOCK_M: tl.constexpr,
+        NUM_CTAS: tl.constexpr,
+    ):
+        # Configure the number of CTAs participating in reduction
+        BLOCK_N: tl.constexpr = triton.cdiv(N, NUM_CTAS)
+
+        # Allocate NUM_CTAS buffers in shared memory, each with shape (BLOCK_M,)
+        # to hold a 1D vector of float32 values
+        local_buffs = tlx.local_alloc((BLOCK_M, ), tl.float32, NUM_CTAS)
+
+        # Allocate barriers for synchronization across CTAs
+        # Each non-zero CTA will use a barrier to signal when its data is written
+        barriers = tlx.alloc_barriers(num_barriers=NUM_CTAS)
```

Comment:
Why did you use NUM_CTAs barriers? Can't you just use one barrier to achieve the sync in this kernel?

Reply:
Yes we can use just one barrier, but then the master CTA would be blocked on all other CTAs to complete, even if some of them can finish earlier. I guess it's up to users.